app/frontend/src/features/profile/edit/EditHostingPreference.test.tsx

```diff
@@ -47,10 +44,43 @@ describe("EditHostingPreference", () => {
   });
 
   it("should redirect to the user profile route after successful update", async () => {
-    renderPage();
+    jest.isolateModules(() => {
+      jest.mock("components/MarkdownInput");
+      const EditHostingPreference = require("./EditHostingPreference").default;
+      renderPage(EditHostingPreference);
+    });
 
     userEvent.click(await screen.findByRole("button", { name: SAVE }));
 
     expect(await screen.findByTestId("user-profile")).toBeInTheDocument();
+  });
+
+  it(`should not submit the default headings for the '${ABOUT_HOME}'section`, async () => {
+    jest.isolateModules(() => {
+      getUserMock.mockImplementation(async (user) => ({
+        ...(await getUser(user)),
+        aboutPlace: "",
+      }));
+      jest.unmock("components/MarkdownInput");
+      const EditHostingPreference = require("./EditHostingPreference").default;
+      renderPage(EditHostingPreference);
+    });
```

Comment:
I was mocking this before as the toast UI editor was quite heavy to render and since interaction with it is a bit dodgy in the tests (keyboard events were firing out of order for some bizarre reasons that I still haven't figured out).

However, by mocking the input, I introduced a different problem... the mock (it's a literal swap to using a bog standard `<input>`) is introducing a different behaviour when the submit button is clicked that doesn't happen with the markdown input. So for this test only, I needed to unmock the markdown component to get the same behaviour. But doing `jest.unmock` has no effect when the component is imported at the top level, since it's already loaded with the mocked version, hence this funky looking setup with `jest.isolateModules`.

Reply:
Some weird shit is happening here, the tests seem to work again with the mocks (as I'd expect in the first place). I hate frontend tests sometimes 😂